food.py

```diff
@@ -0,0 +1,97 @@
+import requests
+import os
+import re
+import time
+import ast
+
+def main():
+    dishes_found = {}
+    yelp_dish = ""
+
+    while dishes_found == {}:
+        dish_input = get_dish()
+        dishes_found = get_dish_info(dish_input)
+
+        if dishes_found == "error":
+            break
+
+        display_result(dishes_found)
+        if dishes_found != {}:
+            yelp_dish = search_restraunt(dishes_found)
+    return yelp_dish
+
+def get_dish():
+    while True:
+        dish_input = input('Enter dish name: ').strip().lower()
+
+        if not dish_input.replace(' ','').isalpha():
+            print("Input was either empty or contained an invalid character, please try again.")
+            continue
+        else:
+            dish_input = re.sub(' +', ' ', dish_input)
+            break
+    return dish_input
+
+def get_dish_info(dish_input):
+    response = request_dishes(dish_input)
+    
+    if response == "error":
+        return response
+    else:
+        dishes_found = convert_response(response)
+        return dishes_found
+
+def request_dishes(dish_input):
+    string = os.environ.get('headers')
+    headers = ast.literal_eval(string)
+    url = "https://rapidapi.p.rapidapi.com/search"
+    querystring = {"q":dish_input}
+
+    for i in range(3):
+        try:
+            response = requests.request("GET", url, headers=headers, params=querystring)
+            if response.status_code != 200:
+                print("Unable to connect to API, retrying in 5 seconds.")
+                time.sleep(5)
+                if i == 3:
+                    response = "error"
+            else:
+                break
+        except requests.HTTPError:
+            print("Service is unavailable or your internet is down. Please try again later.")
+            response = "error"
+            break
+        except Exception:
+            print("An error has occured, please contact our support center.")
+            response = "error"
+            break
+    if response == "error":
```

Comment:
There's a chance the 'response' variable might not be assigned before it gets to here. In that case, the exceptions should keep it from getting this far though.
One solution might be to start with response as 'error' at the beginning of the function and only change it if you get a good response, otherwise just let it ride.
(This is low priority)